src/EPR.Payment.Mopup.Common.Data/Repositories/PaymentsRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using EPR.Payment.Mopup.Common.Constants;
+using EPR.Payment.Mopup.Common.Data.Interfaces;
+using EPR.Payment.Mopup.Common.Data.Interfaces.Repositories;
+using EPR.Payment.Mopup.Common.Enums;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+
+namespace EPR.Payment.Mopup.Common.Data.Repositories
+{
+    public class PaymentsRepository : IPaymentsRepository
+    {
+        private readonly IAppDbContext _dataContext;
+        private readonly IConfiguration _configuration;
+        public PaymentsRepository(
+            IAppDbContext dataContext,
+            IConfiguration configuration)
+        {
+            _dataContext = dataContext;
```

Comment:
Add null check

Reply:
datacontext can never be null. Happy to discuss

Reply:
I agree with Manish but I added it just in case